_posts/2017/2017-10-29-use-simplecov.markdown

```diff
@@ -0,0 +1,67 @@
+---
+title: 'Use simplecov in your Rails test suite'
+description: 'It’s not a silver bullet, but it is useful.'
+layout: article
+category: ruby
+image:
+  base: '2017/use-simplecov'
+  alt: 'Coverage HTML output'
+---
+
+A useful way of assessing the effectiveness of your testing is to use the [`simplecov` gem](https://github.com/colszowka/simplecov) with your Rails application to measure the lines of code that are exercised during your test suite.
+
+## Use…
+
+…`simplecov` in your tests.
+
+#### Add to `Gemfile`
+
+```ruby
+group :test do
+  gem 'simplecov', require: false
+end
+```
+
+And run `bundle` in the root directory of your project
+
+#### Add to `.gitignore`
+
+Do not commit generated files in the `coverage` directory.
+
+```
+coverage/*
+```
+
+#### At the top of `spec/spec_helper.rb` or `test/test_helper.rb`
+
+```ruby
+require 'simplecov'
+SimpleCov.start 'rails' do
+  add_filter '/bin/'
+  add_filter '/db/'
+  add_filter '/spec/' # for rspec
+  add_filter '/test/' # for minitest
+end
+```
+Add this right at the top of the file. The `add_filter` lines means files matching the passed string are excluded from the results.
+
+Then run your tests as normal and then open the `coverage/index.html` file that is created and you can explore your code.
+
+
+## But why?
+
+Measuring the coverage of your application’s code by your test suite is a blunt instrument. In fact, aiming for 100% coverage is not a noble aim, or even useful.
```

Comment:
Mmmm...I see what you're trying to say but it doesn't flow naturally from the "But why?"

Maybe start with some of the stuff you've got later down like: "Having an awareness..." and then later on talk about how aiming for 100% coverage is not a useful aim/not the point of the exercise.